### ISSUE TITLE
Rethink the library

### ISSUE THREAD

**Issue by ghidello** [sotsera/sotsera.blazor.oidc#15]
Update
--------
On hold  waiting for dotnet/aspnetcore#17014 and specifically for dotnet/aspnetcore#18783

-------

Some of the main focus points

* [ ] Allow multiple clients for multiple identity providers
* [ ] Add the user claims to the ClaimsPricipal so they can be used by the policies system. Consider dropping the User class
* [ ] ~~Use Microsoft.IdentityModel.JsonWebTokens~~ (it adds 2.5MB uncompressed and 1.1MB compressed. Skipping for now)
* [ ] Better customization for Authentication and Logout callbacks
  * [ ] Maybe using events instead of Actions
  * [ ] Allow local only logout for Idps
* [ ] Better handling for Idps not supporting the OpenID Connect Session Management 1.0 protocol
  * [ ] Warning for session management not supported
  * [ ] Custom remote logout
* [ ] Opions to consider
  * [ ] Ability to persist the server metadata info
  * [ ] Delayed client configuration (as suggested by @brettwinters)

**Comment by brettwinters** [comment]
All good!

One thing that would be nice (but much less common use case) is for multi-tenant support. I'm thinking about a workflow where the Blazor client (after asking for the user's email address or getting it from the hostname) would fetch user's IDP metadata for their tenant from a back end service then config sotsera.blazor.oidc dynamically...

**Comment by ghidello** [comment]
@brettwinters good point. The configuration will have to be stored in the local or session storage for supporting the redirect flows. I'll think about it.
